amethyst_renderer_new/src/memory.rs

```diff
@@ -0,0 +1,151 @@
+use std::sync::Arc;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use gfx_hal::buffer::{CreationError, Usage, ViewError};
+use gfx_hal::device::{BindError, OutOfMemory};
+use gfx_hal::mapping::Error as MappingError;
+use gfx_hal::memory::Properties;
+use gfx_hal::{Device as GfxDevice, MemoryType};
+
+use back::*;
+
+error_chain! {
```

Comment:
:+1: :+1: I scanned the `error_chain!` source today and I think it's something I want to have in Amethyst.